internal/app/init/pkg/system/conditions/conditions_test.go

```diff
@@ -4,11 +4,160 @@
 
 package conditions_test
 
-import "testing"
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
 
-func TestEmpty(t *testing.T) {
-	// added for accurate coverage estimation
-	//
-	// please remove it once any unit-test is added
-	// for this package
+	"github.com/stretchr/testify/suite"
+
+	"github.com/talos-systems/talos/internal/app/init/pkg/system/conditions"
+)
+
+type ConditionsSuite struct {
+	suite.Suite
+
+	tempDir string
+}
+
+func (suite *ConditionsSuite) SetupSuite() {
+	var err error
+	suite.tempDir, err = ioutil.TempDir("", "talos")
+	suite.Require().NoError(err)
+
+}
+
+func (suite *ConditionsSuite) TearDownSuite() {
+	suite.Require().NoError(os.RemoveAll(suite.tempDir))
+}
+
+func (suite *ConditionsSuite) createFile(name string) (path string) {
+	path = filepath.Join(suite.tempDir, name)
+	f, err := os.Create(path)
+	suite.Require().NoError(err)
+
+	suite.Require().NoError(f.Close())
+
+	return
+}
+
+func (suite *ConditionsSuite) TestFileExists() {
+	exists, err := conditions.FileExists("no-such-file")(context.Background())
+	suite.Require().NoError(err)
+	suite.Require().False(exists)
+
+	exists, err = conditions.FileExists(suite.createFile("a.txt"))(context.Background())
+	suite.Require().NoError(err)
+	suite.Require().True(exists)
+}
+
+func (suite *ConditionsSuite) TestWaitForFileToExist() {
+	path := suite.createFile("w.txt")
+
+	exists, err := conditions.WaitForFileToExist(path)(context.Background())
+	suite.Require().NoError(err)
+	suite.Require().True(exists)
+
+	suite.Require().NoError(os.Remove(path))
+
+	errCh := make(chan error)
+
+	go func() {
+		exists, err = conditions.WaitForFileToExist(path)(context.Background())
+		suite.Require().True(exists)
+
+		errCh <- err
+	}()
+
+	time.Sleep(50 * time.Millisecond)
+
+	select {
+	case <-errCh:
+		suite.Fail("unexpected return")
+	default:
+	}
+
+	suite.createFile("w.txt")
+
+	suite.Require().NoError(<-errCh)
+
+	suite.Require().NoError(os.Remove(path))
+	ctx, ctxCancel := context.WithCancel(context.Background())
+
+	go func() {
+		_, err = conditions.WaitForFileToExist(path)(ctx)
+		errCh <- err
+	}()
+
+	time.Sleep(50 * time.Millisecond)
+
+	select {
+	case <-errCh:
+		suite.Fail("unexpected return")
+	default:
+	}
+
+	ctxCancel()
+
+	suite.Require().EqualError(<-errCh, context.Canceled.Error())
+}
+
+func (suite *ConditionsSuite) TestWaitForFilesToExist() {
+	pathA := suite.createFile("wA.txt")
+	pathB := suite.createFile("wB.txt")
+
+	exists, err := conditions.WaitForFilesToExist(pathA, pathB)(context.Background())
+	suite.Require().NoError(err)
+	suite.Require().True(exists)
+
+	suite.Require().NoError(os.Remove(pathB))
+
+	errCh := make(chan error)
+
+	go func() {
```

Comment:
I'm not sure I follow why we're running this test again in a goroutine? It's the same check we're running on L112

Reply:
we have two conditions here, one is `WaitForFileToExist`, and another one is `WaitForFilesToExist`, so I wanted to cover both of them after my changes (single/multiple). 

I thought about adding generic multiplexed condition `WaitFor(conditions ...ConditionFunc)` so that we can have `WaitFor(FileToExist(/path/a), ServiceToBeStarted("containerd"))`, but that might be future improvement

Reply:
A generic multiplexed condition sounds like an excellent idea. We might even be able to collapse `WaitForFileToExist` into `WaitForFilesToExist`. Running `WaitForFilesToExist` with one file is the same as `WaitForFileToExist`. But this we can do in another PR.

Reply:
yep, exactly!